File: beginner_contest_169/D_Div_Game/v1.cpp
#include<iostream>
#include<vector>
#include<utility>
using namespace std;

vector<pair<long, int>> primeFactorize(long n) {
    vector<pair<long, int>> primes;
    for (long p = 2; p * p <= n; p++) {
        if (n % p != 0) continue;

        int e = 0;
        while (n % p == 0) {
            n /= p;
            e++;
        }
        primes.push_back(make_pair(p, e));
    }

    if (n != 1) primes.push_back(make_pair(n, 1));
    return primes;
}

int main() {
    long n;
    cin >> n;

    int count = 0;
    vector<pair<long, int>> primes(primeFactorize(n));
    for (const pair<long, int>& prime: primes) {
        long p = prime.first;
        int e = prime.second;
        for (int i = 1; i <= e; i++) {
            e -= i;
            count += 1;
        }
    }
    cout << count;
}

File: beginner_contest_169/D_Div_Game/v1.js
const fs = require('fs')
const n = parseInt(fs.readFileSync('/dev/stdin', 'utf8').trim())

function primeFactorize(n) {
  const primes = []
  for (let p = 2; p * p <= n; p += 1) {
    if (n % p != 0) continue

    let e = 0
    while (n % p == 0) {
      n /= p
      e += 1
    }
    primes.push({ p, e })
  }

  if (n != 1) primes.push({ p: n, e: 1 })
  return primes
}

let count = 0
const primes = primeFactorize(n)
primes.forEach(prime => {
  let e = prime.e
  for (let i = 1; i <= e; i += 1) {
    e -= i;
    count += 1;
  }
})

console.log(count)
